### PR TITLE
Adds profiles + settings ui

### LINKED ISSUE #56: Profile | log in
child of guestlogix/traveler-ios#55

Depends on guestlogix/guest#215

As a GLX team member I want to log in using my guestlogix gmail account  so that I can have secure access to the GCrew platform

**Acceptance Criteria**
1. GLX team member can log in from the profile page with the MVP
1. GLX team member must have successfully passed the Gmail authentication
1. On successful log in, the user remains on the profile page and will see the logged in state
1. A user stays logged in until manually logging out
1. Only GLX email domains are accepted

**Out of scope/Future consideration**
Any relevant out of scope items

**UX if needed**
***Designs:***
https://zpl.io/aXP57Kj
*Tags: Profile*

**Technical details if needed**

### Request
_e.g. `GET /example/{id}?query=xyz&also=xyz`

**Body**
```
{ "test : "this" }
```

### Response
**Body**
```
{ "test : "this" }
```

**Status Codes**
_e.g. 200 Success_

## Service
_e.g. Business logic in this layer_

## Repository
_e.g. Data access logic in this layer_

## Database
_e.g. Database specific concerns_

**Tables**
_e.g. Account_Traveler_

**Columns**
_e.g. Date Created_

**Testing considerations**
1. No concerns with syncing GLX gmail passwords.  ie: if i change the password my secure session within the gcrew app remains

### LINKED ISSUE #163: Profile | View profile - MVP
child of guestlogix/traveler-ios#55

As a traveler I want see my user profile so that I can confirm my details available within the app

**Acceptance Criteria**
1. The profile is accessible from the home page
1. The profile will display the travelers name, email address ~, and saved payment info if available~
~1. There is a link for editing the profile~
1. There's a link to order page
1. There is a link for the app settings

**Out of scope/Future consideration**
Any relevant out of scope items

**UX if needed**
- https://app.zeplin.io/project/5bc0d5e9e2c97b18e1537de2/dashboard?tag=Home%20page
- https://app.zeplin.io/project/5bc0d5e9e2c97b18e1537de2/dashboard?tag=Profile

**Technical details if needed**

### Request
_e.g. `GET /example/{id}?query=xyz&also=xyz`

**Body**
```
{ "test : "this" }
```

### Response
**Body**
```
{ "test : "this" }
```

**Status Codes**
_e.g. 200 Success_

## Service
_e.g. Business logic in this layer_

## Repository
_e.g. Data access logic in this layer_

## Database
_e.g. Database specific concerns_

**Tables**
_e.g. Account_Traveler_

**Columns**
_e.g. Date Created_

**Testing considerations**

### LINKED ISSUE #165: Profile | View app settings - MVP
child of guestlogix/traveler-ios#55

As a traveler I want see application information so that I can have better understanding of what is available in the app and how to use it

**Acceptance Criteria**
1. Accessible from the profile page
1. Display link to Legal information **Need COPY**
1. Display link to Privacy Policy **Need Copy**
1. Display trigger to sign out of profile 
~1. Display trigger to deleting profile~
1. Display app version

**Out of scope/Future consideration**
Any relevant out of scope items

**UX if needed**
- https://app.zeplin.io/project/5bc0d5e9e2c97b18e1537de2/dashboard?tag=Profile

**Technical details if needed**

### Request
_e.g. `GET /example/{id}?query=xyz&also=xyz`

**Body**
```
{ "test : "this" }
```

### Response
**Body**
```
{ "test : "this" }
```

**Status Codes**
_e.g. 200 Success_

## Service
_e.g. Business logic in this layer_

## Repository
_e.g. Data access logic in this layer_

## Database
_e.g. Database specific concerns_

**Tables**
_e.g. Account_Traveler_

**Columns**
_e.g. Date Created_

**Testing considerations**

### PR DESCRIPTION
**Pull request by dorothyfu**
**Description**
- The profile + settings screen does not follow the mocks 100%, but I spoke to @Do-Jones about the modifications already. Should follow https://app.zeplin.io/project/5bc0d5e9e2c97b18e1537de2/screen/5c50bcd9188fc5016c5445a6
- Currently DOES NOT persist profile data on the app (e.g. data is not saved into userDefaults). 
- Does not include edit profile
- I left the profile picture placeholder in, but can be removed since profile pictures are not part of this scope 

**Issues that should be CLOSED by merge of this PR:**
* Fixes #165 
* Fixes #163 
* Fixes #56 

**Related issues that should be LINKED to this PR:**
* Connects

### CODE REVIEW

File: Traveler/ViewControllers/ProfileViewController.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
+//
+//  ProfileViewController.swift
+//  Traveler
+//
+//  Created by Dorothy Fu on 2019-03-19.
```
Comment (Skrinikov) [inline Traveler/ViewControllers/ProfileViewController.swift]: Oof

File: Traveler/ViewControllers/ProfileViewController.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch (indexPath.section, indexPath.row) {
+        case (0, _):
+            let cell = tableView.dequeueReusableCell(withIdentifier: "profileCellIdentifier", for: indexPath) as! ProfileCell
+            if profile?.firstName != nil && indexPath.row == 0 {
+                cell.titleLabel.text = "Name"
+                cell.valueLabel.text = "\(profile?.firstName ?? "")\(profile?.lastName ?? "")"
```
Comment (Skrinikov) [inline Traveler/ViewControllers/ProfileViewController.swift]: No space between fName and lastName?
DorothyFu

File: Traveler/ViewControllers/SettingsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+extension SettingsViewController {
+    func showAlert(withTitle title: String, withMessage message:String) {
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        let ok = UIAlertAction(title: "OK", style: .default, handler: { action in
+            self.logOut()
+            self.performSegue(withIdentifier: "exitSegue", sender: self)
+        })
+        let cancel = UIAlertAction(title: "Cancel", style: .default, handler: { action in
+        })
```
Comment (Skrinikov) [inline Traveler/ViewControllers/SettingsViewController.swift]: can this be on the same line?

File: Traveler/Views/ProfileCell.swift [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+//  Created by Dorothy Fu on 2019-03-20.
+//  Copyright © 2019 GuestLogix. All rights reserved.
+//
+
+import UIKit
+
+class ProfileCell: UITableViewCell {
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var valueLabel: UILabel!
+
```
Comment (Skrinikov) [inline Traveler/Views/ProfileCell.swift]: extra line

File: Traveler/ViewControllers/ProfileViewController.swift (hunk 2 above)
Comment (dorothyfu) [reply Traveler/ViewControllers/ProfileViewController.swift]: It comes with a trailing space on the first name @Skrinikov

File: Traveler/ViewControllers/SettingsViewController.swift (hunk 3 above)
Comment (dorothyfu) [reply Traveler/ViewControllers/SettingsViewController.swift]: fine

File: Traveler/ViewControllers/ProfileViewController.swift (hunk 1 above)
Comment (GeorgiosP) [reply Traveler/ViewControllers/ProfileViewController.swift]: oof

File: Traveler/ViewControllers/ProfileViewController.swift (hunk 2 above)
Comment (Skrinikov) [reply Traveler/ViewControllers/ProfileViewController.swift]: Well thats intuitive

Comment (dorothyfu) [reply Traveler/ViewControllers/ProfileViewController.swift]: @GeorgiosP will trim extra white space and iOS + Android will add a space

**Pull request merged**